common/src/main/java/com/wynntils/commands/LocateCommand.java

```diff
@@ -0,0 +1,176 @@
+                        .then(Commands.argument("name", StringArgumentType.greedyString())
+                                .suggests(SERVICE_SUGGESTION_PROVIDER)
+                                .executes(this::locateService))
+                        .build())
+                .then(Commands.literal("place")
+                        .then(Commands.argument("name", StringArgumentType.greedyString())
+                                .suggests(PLACES_SUGGESTION_PROVIDER)
+                                .executes(this::locatePlace))
+                        .build())
+                .then(Commands.literal("npc")
+                        .then(Commands.argument("name", StringArgumentType.greedyString())
+                                .executes(this::notImplemented))
+                        .build())
+                .then(Commands.literal("other")
+                        .then(Commands.argument("name", StringArgumentType.greedyString())
+                                .executes(this::notImplemented))
+                        .build())
+                .executes(this::syntaxError);
+    }
+
+    private int locateService(CommandContext<CommandSourceStack> context) {
+        String searchedName = context.getArgument("name", String.class);
+
+        List<ServiceKind> matchedKinds = Arrays.stream(ServiceKind.values())
+                .filter(kind -> StringUtils.partialMatch(kind.getName(), searchedName))
+                .toList();
+
+        if (matchedKinds.isEmpty()) {
+            MutableComponent response = new TextComponent("Found no services matching '" + searchedName + "'")
+                    .withStyle(ChatFormatting.RED);
+            context.getSource().sendFailure(response);
+            return 0;
+        }
+
+        if (matchedKinds.size() > 1) {
+            MutableComponent response = new TextComponent("Found multiple services matching '" + searchedName
+                            + "'. Pleace specify with more detail. Matching: ")
+                    .withStyle(ChatFormatting.RED);
+            response.append(new TextComponent(String.join(
+                    ", ", matchedKinds.stream().map(kind -> kind.getName()).toList())));
+            context.getSource().sendFailure(response);
+            return 0;
+        }
+
+        ServiceKind selectedKind = matchedKinds.get(0);
+
+        List<Poi> services = new ArrayList<>(MapModel.getAllPois().stream()
+                .filter(poi -> poi instanceof ServicePoi servicePoi
+                        && servicePoi.getKind().equals(selectedKind))
+                .toList());
+
+        // Only keep the closest results
+        Vec3 currentLocation = McUtils.player().position();
+        services.sort(Comparator.comparingDouble(poi -> currentLocation.distanceToSqr(
+                poi.getLocation().getX(),
+                poi.getLocation().getY(),
+                poi.getLocation().getZ())));
+        services.subList(4, services.size()).clear();
```

Comment:
What is this?

Reply:
Remove everything except the three first elements in the list. As suggested by Stack Overflow.

Reply:
Ah.., makes sense. Maybe I would have figured out after staring for like 10 minutes. I just spent 1 hour wondering why my POI rendering trigonometry did not work... (I centered my math around the screen and not the map center.... :facepalm: )